packages/strapi-generate-new/lib/merge-template.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const path = require('path');
+const fse = require('fs-extra');
+const fetch = require('node-fetch');
+const unzip = require('unzip-stream');
+const _ = require('lodash');
+
+// Specify all the files and directories a template can have
+const allowChildren = '*';
+const allowedTemplateTree = {
+  // Root template files
+  'template.json': true,
+  'README.md': true,
+  '.gitignore': true,
+  // Template contents
+  template: {
+    'README.md': true,
+    '.env.example': true,
+    api: allowChildren,
+    components: allowChildren,
+    config: {
+      functions: allowChildren,
+    },
+    data: allowChildren,
+    plugins: allowChildren,
+    public: allowChildren,
+    scripts: allowChildren,
+  },
+};
+
+// Traverse template tree to make sure each file and folder is allowed
+async function checkTemplateStructure(templatePath) {
```

Comment:
We should allow extra files in the repositories. A developer might want to setup some tooling to create there templates. (e.g, prettifing the files, linting them etc) We shoul just make sure the template.json and template folder exist add have the write structure. outside of that we can leave more freedom

Reply:
I think that would mean that we'd have more of a blacklist system, instead of a whitelist. Is that right?

As in, allow everything in the `template/` directory, except `config/database.js` and other files that would cause conflicts with the CLI options

Reply:
Actually I was only saying we should allow extra fiels at the root of the projects and restrict template.json and template folder ;)